execute/executor.go

```diff
@@ -287,17 +300,49 @@ func (v *createExecutionNodeVisitor) Visit(node plan.Node) error {
 				}
 			}
 
-			if (plan.HasSideEffect(spec) || isParallelMerge) && len(node.Successors()) == 0 {
-				name := string(node.ID())
-				r := newResult(name)
-				v.es.results[name] = r
-				v.nodes[skipYields(node)][i].AddTransformation(r)
+			// XXX: results should be generated for terminal nodes, regardless
```

Comment:
Unsure if/when this would not hold true. It could make sense to therefore hardcode the 0th index into the function body instead of exposing it as a parameter, but I guess it's fine as is.

Nice to move it to one spot! Being able to remove a comment to explain something awkward feels good.